cmd/compose

```diff
@@ -188,23 +188,25 @@ gen_certs() {
     ####
     # Fresh start
     #### 
-    TEMP_CERT_PATH="${HOME}/docker_certs"
-    rm -rf ${TEMP_CERT_PATH}
-    mkdir -p ${TEMP_CERT_PATH}
+    TEMP_CERT_PATH=$(mktemp -d tmp.XXXXXXXXXX) || { echo "Failed to create temp file"; exit 1; }
+    TEMP_DOCKER_MACHINE_CERTS=$(mktemp -d) || { echo "Failed to create temp file"; exit 1; }
+    DOCKER_MACHINE_CERTS=${HOME}/.docker/machine/certs
+    GEN_CERTS_LOG="GEN_CERTS_`date '+%y%m%d-%H%M%S'`.log"
+
+    #Copying the docker machine certificates to temporary folder
+    cp ${DOCKER_MACHINE_CERTS}/* ${TEMP_DOCKER_MACHINE_CERTS}
 
     ####
     # * Generate the client private key
     # * Generate signed certificate
     # * Generate the client certificate
     ####
-    set +e
-    openssl genrsa -out ${TEMP_CERT_PATH}/key.pem 4096 &> /dev/null
-    openssl req -subj "${CLIENT_SUBJ}" -new -key ${TEMP_CERT_PATH}/key.pem -out ${TEMP_CERT_PATH}/client.csr &> /dev/null
-    echo "extendedKeyUsage = clientAuth" >  ${TEMP_CERT_PATH}/extfile.cnf
-    openssl x509 -req -days 365 -sha256 -in ${TEMP_CERT_PATH}/client.csr -CA ${HOME}/.docker/machine/certs/ca.pem -CAkey ${HOME}/.docker/machine/certs/ca-key.pem -CAcreateserial -CAserial temp.seq -out ${TEMP_CERT_PATH}/cert.pem -extfile ${TEMP_CERT_PATH}/extfile.cnf &> /dev/null
-    set -e
-    cp ${HOME}/.docker/machine/certs/ca.pem ${TEMP_CERT_PATH}/ca.pem
-    docker --tlsverify --tlscacert=${HOME}/.docker/machine/certs/ca.pem --tlscert=${TEMP_CERT_PATH}/cert.pem --tlskey=${TEMP_CERT_PATH}/key.pem -H=${DOCKER_HOST} version &> /dev/null
+    openssl genrsa -out ${TEMP_CERT_PATH}/key.pem 4096 > ${GEN_CERTS_LOG} 2>&1 || { echo "Error generating certificates. Please see " ${GEN_CERTS_LOG} " for more info"; exit 1; }
```

Comment:
@anton-kasperovich , it is the first time we are writing to this file, so I don't think it matters whether we are using > or >> . In fact, in this case we are making sure that we are starting with an empty log file. What do you think?